sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/InvitationService.java

```diff
@@ -150,27 +153,34 @@ public Invitation retryInvitation(Long principalId, String email) {
         invitationFromDb.setCreatedBy(userService.getUserById(principalId));
         invitationFromDb = updateInvitation(invitationFromDb);
         sendEmail(invitationFromDb);
+
+        insertInvitationUrl(invitationFromDb);
         return invitationFromDb;
     }
 
     @Transactional(readOnly = true)
     public Invitation getInvitationByToken(String token) {
-        return invitationMapper.getInvitationByCode(token);
+        Invitation invitation = invitationMapper.getInvitationByCode(token);
+        insertInvitationUrl(invitation);
+        return invitation;
     }
 
     @Transactional(readOnly = true)
     public Invitation getInvitationByEmail(String email) {
-        return invitationMapper.getInvitationByEmail(email);
+        Invitation invitation = invitationMapper.getInvitationByEmail(email);
+        insertInvitationUrl(invitation);
+        return invitation;
     }
 
     @Transactional(readOnly = true)
     public List<Invitation> getAllInvitations() {
-        return invitationMapper.getAllInvitations();
+        return invitationMapper.getAllInvitations().stream().peek(this::insertInvitationUrl).collect(Collectors.toList());
```

Comment:
Please format so every new call on stream will start on a new line for the sake of readability